mathics/builtin/numeric.py

```diff
@@ -863,24 +878,52 @@ class N(Builtin):
         ),
     }
 
-    rules = {
-        "N[expr_]": "N[expr, MachinePrecision]",
-    }
-
     summary_text = "numerical evaluation to specified precision and accuracy"
 
-    def apply_with_prec(self, expr, prec, evaluation):
-        "N[expr_, prec_]"
+    def apply_with_prec(self, expr, prec, evaluation, options=None):
+        "N[expr_, prec_, OptionsPattern[%(name)s]]"
+
+        # If options are passed, set the preference in evaluation, and call again
+        # without options set.
+        # This also prevents to store this as an nvalue (nvalues always have two leaves).
```

Comment:
Where in either WL docs or in this code are nvalues described?

Reply:
![image](https://user-images.githubusercontent.com/2862288/151593318-c158d63b-1948-4463-82ae-ae9b5f027a1f.png)


For some reason, the documentation of this symbol was removed from WR.

Reply:
Maybe for stuff like this we should start a "compatibility" package for Mathematica 5 so that if this is what you want you need to ask for it explicitly.

Reply:
Okay it's there. But we say that it has been removed then when it isn't!

Reply:
https://reference.wolfram.com/language/tutorial/TransformationRulesAndDefinitions.html  section "Defining Numerical Values":

"The Wolfram Language treats numerical values essentially like upvalues. When you define a numerical value for f, the Wolfram Language effectively enters your definition as an upvalue for f with respect to the numerical evaluation operation N. "

Reply:
> Maybe for stuff like this we should start a "compatibility" package for Mathematica 5 so that if this is what you want, you need to ask for it explicitly.

OK, but I am not sure that is was "removed" and kept just for compatibility.  I think it becomes something "internal". Still, if you evaluate
```
N[f[x]]:=3
```
the rule is not stored as an ``Upvalue`` but as a ``NValue``. What is not exposed in the documentation is the keyword ``NValues``.

Reply:
> https://reference.wolfram.com/language/tutorial/TransformationRulesAndDefinitions.html section "Defining Numerical Values":
> 
> "The Wolfram Language treats numerical values essentially like upvalues. When you define a numerical value for f, the Wolfram Language effectively enters your definition as an upvalue for f with respect to the numerical evaluation operation N. "

Interesting. When we write a description of the evaluation process we need to make sure to add links to things like this. Notice they didn't call this an NValue but "define a numerical value", and I am not certain what that means.

Reply:
I think that what currently is not "published" by the kernel (documentation) is the access to these "NValues". However, it makes a lot of sense to store them in a different place, because are rules that are not supposed to be applied in a general evaluation.